key.core/src/main/java/de/uka/ilkd/key/control/KeYEnvironment.java

```diff
@@ -28,7 +28,7 @@
  *
  * @author Martin Hentschel
  */
-public class KeYEnvironment<U extends UserInterfaceControl> {
+public class KeYEnvironment<U extends UserInterfaceControl> implements AutoCloseable {
```

Comment:
I also do not see a great benefit there as anyone can build a wrapper easily:

```
record PA(Proof p) implements AutoCloseable { public void close() {p.dispose();}}
static PA disposeAfter(Proof p) { return new PA(p); }
```

Is it ensured, that `#dispose` is also working on broken proof states (in case of an exception)?

Reply:
Also, we might want to avoid all interface changes in minor updates.

Reply:
I have removed the new implementation for now. The benefit would be that no wrapper is necessary.

Reply:
> I also do not see a great benefit there as anyone can build a wrapper easily:
> 
> ```
> record PA(Proof p) implements AutoCloseable { public void close() {p.dispose();}}
> static PA disposeAfter(Proof p) { return new PA(p); }
> ```
> 
> Is it ensured, that `#dispose` is also working on broken proof states (in case of an exception)?

I do not understand the idea with the wrapper. From my point of view, the benefit in having KeYEnvironment implement the AutoCloseable interface is that you can use it in try-with-resources and usually get a warning from your IDE if you don't. This is an improvement over the current state, since at the moment I am quite sure that there are many places where the resources are not properly cleaned simply because developers did not know or forgot that dispose() should be called.
I can not see how a wrapper would help here.